velox/exec/tests/StreamingAggregationTest.cpp

```diff
@@ -538,3 +616,36 @@ TEST_F(StreamingAggregationTest, distinctAggregations) {
   testMultiKeyDistinctAggregation(multiKeys, 1024);
   testMultiKeyDistinctAggregation(multiKeys, 3);
 }
+
+TEST_F(StreamingAggregationTest, lambdaAggregations) {
```

Comment:
lambdaAggregations -> reduce_agg

Would it make sense to move this test to ReduceAggTest and modify AggregationTestBase::testAggregations to including testing of streaming aggregation plans?

Reply:
Got it, will enhance this part.

Reply:
@mbasmanova Hi Masha, I have to add an `orderBy` node in the plan builder in `testStreamingAggregationsImpl` to include testing of `StreamingAggregation` plans in `AggregationTestBase::testAggregations`. But this would break lots of test cases, say `ArbitraryTest.noNulls`, `ArbitraryTest.varchar`, `MapAggTest.groupByWithDuplicates` and lots of spark agg tests.

I've investigated the failure of `ArbitraryTest.varchar` and `MapAggTest.groupByWithDuplicates` and found the **assert failure caused by the inserted `orderBy` node**, but both results should be the right results. I have two suggestions,
1. Add a testStreamingAggregation flag (false by default) in `AggregationTestBase::testAggregations` and set it as true in the suitable test cases then it would call the `testStreamingAggregationsImpl` method when the grouping keys are not empty.
2. Add a `AggregationTestBase::testStreamingAggregations` API directly and use it when we want to test StreamingAggregation.

What do you think?

## Some Logs
### MapAggTest.groupByWithDuplicates
The two results should be right
```
284: [ RUN      ] MapAggTest.groupByWithDuplicates
284: /root/project/velox/exec/tests/utils/QueryAssertions.cpp:1129: Failure
284: Failed
284: Expected 5, got 5
284: 2 extra rows, 2 missing rows
284: 2 of extra rows:
284:    0 | [{"key":0,"value":1.05}]
284:    2 | [{"key":2,"value":5.05}]
284: 
284: 2 of missing rows:
284:    0 | [{"key":0,"value":0.05}]
284:    2 | [{"key":2,"value":4.05}]
```

### ArbitraryTest.varchar
```
[ RUN      ] ArbitraryTest.varchar
/root/project/velox/exec/tests/utils/QueryAssertions.cpp:1129: Failure
Failed
Expected 22, got 22
22 extra rows, 22 missing rows
10 of extra rows:
        null | "dr"
        -10 | "ijzx"
        -9 | "tudmdrik"
        -8 | "adrey"
        -7 | "vdehxhcp"
        -6 | "yzwukpmyo"
        -5 | "ulgur"
        -4 | "fhkjibfey"
        -3 | "ufyw"
        -2 | "txk"

10 of missing rows:
        null | "cmytq"
        -10 | "vmhoivn"
        -9 | "mtqevd"
        -8 | "rkze"
        -7 | "drlgilljvo"
        -6 | "j"
        -5 | "zpqswck"
        -4 | "tvviehlmq"
        -3 | "sixkd"
        -2 | "gyzvrxyp"
```